sklearn/metrics/tests/test_pairwise.py

```diff
@@ -540,6 +540,18 @@ def test_linear_kernel():
     assert_array_almost_equal(K.flat[::6], [linalg.norm(x) ** 2 for x in X])
 
 
+def test_linear_kernel_sparse_output():
```

Comment:
This is very redundant with `test_cosine_similarity_sparse_output`. 
Let's just make this latter test a parametric test with pytest, that takes as input the similarity function (and the metric name),  to check both for cosine similarity and the linear kernel. Also possibly rename that test to something more generic.

Reply:
No what I meant is to define a single function and make it parametric with pytest,
```py
from functools import partial

@pytest.mark.parametrized('name, pairwise_func',
                          [('linear_kernel', linear_kernel),
                           ('cosine_similarity', cosine_similarity)])
def test_cosine_similarity_sparse_output(name, pairwise_func):
    # adapt the previous version of test_cosine_similarity_sparse_output here
```

see https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions